packages/expo-cli/src/commands/webhooks.ts

```diff
@@ -1,150 +1,171 @@
+) {
+  invariant(typeof id === 'string', '--id must be a webhook ID');
+  invariant(event == null || typeof event === 'string', '--event: string is required');
+  let secret = validateSecret(options);
+
+  const { project, client } = await setupAsync(projectRoot);
+
+  const webhook = await client.getAsync(`projects/${project.id}/webhooks/${id}`);
+  event = event ?? webhook.event;
+  secret = secret ?? webhook.secret;
+
+  const spinner = ora(`Updating webhook ${id}`).start();
+  await client.patchAsync(`projects/${project.id}/webhooks/${id}`, { url, event, secret });
+  spinner.succeed();
+}
 
-  if (!WEBHOOK_TYPES.includes(event)) {
-    throw new Error(`Unsupported webhook type: ${event}`);
-  }
+async function removeAsync(projectRoot: string, { id }: { id?: string }) {
+  invariant(typeof id === 'string', '--id must be a webhook ID');
+  const { project, client } = await setupAsync(projectRoot);
 
-  return event;
+  await client.deleteAsync(`projects/${project.id}/webhooks/${id}`);
 }
 
-async function _askForSecret(): Promise<string> {
-  const { secret } = await inquirer.prompt({
-    type: 'password',
-    name: 'secret',
-    message: 'Webhook secret (at least 16 and not more than 1000 characters):',
-  });
-  if (secret.length < 16 || secret.length > 1000) {
-    log.error('Webhook secret has be at least 16 and not more than 1000 characters long');
-    return await _askForSecret();
-  } else {
+function validateSecret({ secret }: { secret?: string }): string | null {
+  if (secret) {
+    invariant(
+      secret.length >= SECRET_MIN_LENGTH && secret.length < SECRET_MAX_LENGTH,
+      `--secret: should be ${SECRET_MIN_LENGTH}-${SECRET_MAX_LENGTH} characters long`
+    );
     return secret;
   }
+  return null;
+}
+
+function generateSecret() {
+  // Create a 60 characters long secret from 30 random bytes.
+  const randomSecret = crypto.randomBytes(30).toString('hex');
+  log(chalk.underline('Webhook signing secret:'));
+  log(randomSecret);
+  return randomSecret;
+}
+
+async function setupAsync(projectRoot: string) {
+  const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+  const { slug } = exp;
+  if (!slug) {
+    throw new CommandError(
+      ErrorCodes.MISSING_SLUG,
+      `expo.slug is not defined in ${findConfigFile(projectRoot).configName}`
+    );
+  }
+  const user = await UserManager.ensureLoggedInAsync();
+  const client = ApiV2.clientForUser(user);
+  const experienceName = `@${exp.owner ?? user.username}/${exp.slug}`;
+  try {
+    const projects = await client.getAsync('projects', {
+      experienceName,
+    });
+    if (projects.length === 0) {
+      throw projectNotFoundError(experienceName);
+    }
+    const project = projects[0];
+    return { experienceName, project, client };
+  } catch (error) {
+    if (error.code === 'EXPERIENCE_NOT_FOUND') {
+      throw projectNotFoundError(experienceName);
+    } else {
+      throw error;
+    }
+  }
+}
```

Comment:
Write a unit test for this, specifically that we use "owner" if it is specified and fall back to the username otherwise.